src/sdk/namespace_fs.js

```diff
@@ -513,7 +549,37 @@ class NamespaceFS {
                 if (r.common_prefix) {
                     res.common_prefixes.push(r.key);
                 } else {
-                    res.objects.push(this._get_object_info(bucket, r.key, r.stat, 'null'));
+                    const obj_info_list = [];
+                    const obj_info = this._get_object_info(bucket, r.key, r.stat, 'null', true);
+                    if (!list_versions && obj_info.delete_marker) {
+                        continue;
+                    }
+                    if (obj_info.delete_marker) {
+                        obj_info.is_latest = false;
+                    }
+                    if (list_versions) {
+                        versions.forEach(v => {
+                            var search_key = v.key.replace(HIDDEN_VERSIONS_PATH, '');
```

Comment:
can you explain what this code does?

Reply:
Here I am trying to pull the file_path from `version` list by removing `.versions` dir from the file path.

ex:  lets consider that we want to push $BUCKET_PATH/file1 and its versions.

On `.versions` list will have list of all versions of all objects. Here we need to filter `version` list based on the `$BUCKET_PATH/file1`. But the keys on `version` list would have `.versions` in the path name. So to compare the `process_entry()`, we need to remove `.versions` from the file path.

Reply:
in process entry we use _get_entry_key(), can you move this logic there and call _get_entry_key() here instead?